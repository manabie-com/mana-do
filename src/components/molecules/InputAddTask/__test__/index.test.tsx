import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react';
import {Design, InputAddTask} from "../index";

const setup = (value?:string) => {
    const utils = render(<InputAddTask typeCallback="" callback={jest.fn()} value={value} />)
    const input = utils.getByLabelText('todo-input')
    return {
        input,
        ...utils,
    }
}

describe('<InputAddTask /> render', () => {
    it('should match snapshot', function () {
        const Element = render(<InputAddTask typeCallback="" callback={jest.fn()}/>);
        expect(Element.container.firstChild).toMatchSnapshot();
    });
    it('should callback value when keydown Enter', function () {
        const {input} = setup()
        fireEvent.keyDown(input, {key: "Enter"})
        fireEvent.change(input, {current: {value: "123123"}})
        expect(input.nodeValue).toBe('23')
    });
    it('should rerender when change value', function () {
        const {rerender} = render(<InputAddTask typeCallback="" callback={jest.fn()} value="123" design={Design.UPDATE} />)
        expect(screen.getByTestId('todo-input')).toHaveTextContent('1')

        // re-render the same component with different props
        rerender(<InputAddTask callback={jest.fn()} typeCallback="" value="456" design={Design.UPDATE}/>)
        expect(screen.getByTestId('todo-input')).toHaveTextContent('456')
    });
})