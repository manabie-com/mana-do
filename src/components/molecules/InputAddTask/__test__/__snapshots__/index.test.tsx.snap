// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputAddTask /> render should match snapshot 1`] = `
<input
  aria-label="todo-input"
  class="Todo__input"
  data-testid="todo-input"
  placeholder="+ Add a task"
/>
`;
